anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -483,201 +482,240 @@ def image(dbsession, request_inputs, bodycontent=None):
                     raise Exception("image not found in DB")
 
         elif method == "POST":
-            timer = time.time()
+            return_object, httpcode = image_post(
+                userId,
+                input_type,
+                params,
+                bodycontent,
+                dbsession,
+                image_info,
+                input_string,
+            )
+    except AnchoreApiError as err:
+        logger.exception("Error processing image request")
+        return_object = anchore_engine.common.helpers.make_response_error(
+            err.message, in_httpcode=err.__response_code__, details=err.detail
+        )
+        httpcode = err.__response_code__
+    except Exception as err:
```

Comment:
I think we should not allow blanket exceptions anymore as part of new additions https://github.com/anchore/anchore-engine/issues/867

Reply:
I think specifically, I want to leave this error because there are lots of possibilities of errors to be thrown here, but this should be a target for a later refactor

Reply:
Unfortunately that doesn't end up working well for testing, debugging or how the rest of the code-base deals with exceptions. The `image()` function gets called in a couple of places. One is in `list_images()` and the other one in `add_image()`, *both of which* do enormous try/except blocks that are also catching anything that might get raised. 
As I explain in #867, this broad exception catching at multiple levels in the stack makes it impossible to debug, and causes problems for testing.

I'm going to be a -1 on adding more broad-exception catches like this one. There needs to be a solid reason for catching everything, while avoiding multiple broad try/except blocks that are catching exceptions.

Reply:
@alfredodeza I totally agree with you here, but I have two comments:
* I'm not adding this exception block (it's not new), just ported over as I modularized things
* If I change this logic now...I need to go through each sub method, identify _all possible_ errors, and be 100% sure that expected use cases don't end up missing a more specific block. At this point, that's not a risk I'm looking to take as this is a highly used path in the engine. At this point I am just trying to do the bare bones modularization...but you are 100% right that this is bad and needs to change, I'm just trying to avoid doing that change in this PR.